### ISSUE TITLE
🤗 [Question]: QueryParser test coverage problem

### ISSUE THREAD

**Issue by ugurcanerdogan** [gofiber/fiber#2587]
Hello there! Can you give an example wrong input for this method that parses the query params in a request to **return an error**? During unit testing, we cannot cover the cases where this method returns an error, and we have tried many ways. Finally, we decided not to handle the error returned from here. Thanks!

We opened this issue because we thought there might be a problem. 

[Error returning case 1](https://github.com/gofiber/fiber/blob/5fd3f20de8e2e745e1760d921e56ae6796b619ed/ctx.go#L1220C4-L1220C4
)
[Error returning case 2](https://github.com/gofiber/fiber/blob/5fd3f20de8e2e745e1760d921e56ae6796b619ed/ctx.go#L1234C1-L1234C1) 
[Error returning case 3](https://github.com/gofiber/fiber/blob/5fd3f20de8e2e745e1760d921e56ae6796b619ed/ctx.go#L1237)

**Comment by ReneWerner87** [comment]
@ugurcanerdogan 
> we cannot cover the cases where this method returns an error

why would you do that?
covering these parts is part of the library unittests

**Comment by ReneWerner87** [comment]
besides, this error is extremely unlikely, then something must be wrong with the buffers
we only passed on these errors because the underlying functionalities return them and it is common practice in golang to pass them on

**Comment by ReneWerner87** [comment]
i close the report, if there are further questions, reopen this topic or ask in the discord chat

**Comment by ugurcanerdogan** [comment]
> @ugurcanerdogan
> 
> > we cannot cover the cases where this method returns an error
> 
> why would you do that? covering these parts is part of the library unittests

Since the related cases can return errors, we need to handle and cover these cases in our own implementation. Otherwise our coverage percentage decreases. In order not to decrease, we ignore the error that can be returned with the help of _ (underscore) and avoid it in this way. 

The library itself has covered these errors with its unittests, but I'm also not sure why we should cover it, as you said, only our coverage count seems to have dropped due to this situation and after research I realized that some cases were not possible. Then we decided to ignore it.

For example, when the non-obligatory queryParam values are not given during the request, we cannot enter these cases and we never get an error and therefore we cannot cover them in our unittests. The error returned by the code occurs when the enforced queryParam values are not given in the request. Maybe as a suggestion, these values can be checked and whether they return an error or not, depending on whether they are mandatory or not.
